### PR TITLE
Add tasklist to visibility 

### PR DESCRIPTION
**Pull request by mkolodezny**
**What changed?**

Add tasklist to visibility record. 

**Why?**

To provide possibility to search workflows by task list with ES

**How did you test it?**

unit tests
cadence cli 

**Potential risks**

Search queries to ES

### CODE REVIEW

**Comment by yycptt** [comment]
I thought the original issue means include task list as one of the default search attributes (just like `BinaryCheckSums` and `CadenceChangeVersion`), not adding a new field to the visibility record.  Am I missing something here?

I am concerned about this because if we add a new field to the visibility record, it will impact visibility archival, especially our internal implementation.

**Comment by vancexu** [comment]
> I thought the original issue means include task list as one of the default search attributes (just like `BinaryCheckSums` and `CadenceChangeVersion`), not adding a new field to the visibility record. Am I missing something here?
> 
> I am concerned about this because if we add a new field to the visibility record, it will impact visibility archival, especially our internal implementation.

The major reason is to also benefit users without ES.
We will need to continue adding this to visibility archival before release.

**Comment by yycptt** [comment]
> > I thought the original issue means include task list as one of the default search attributes (just like `BinaryCheckSums` and `CadenceChangeVersion`), not adding a new field to the visibility record. Am I missing something here?
> > I am concerned about this because if we add a new field to the visibility record, it will impact visibility archival, especially our internal implementation.
> 
> The major reason is to also benefit users without ES.
> We will need to continue adding this to visibility archival before release.

I see. Let's take the discussion offline and probably explore if there's an alternative. Making changes to visibility archival, especially our internal implementation will be a little bit tricky.

File: common/persistence/elasticsearch/decodeBench_test.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ var (
          "WorkflowID": "6bfbc1e5-6ce4-4e22-bbfb-e0faa9a7a604-1-2256",
          "WorkflowType": "TestWorkflowExecute",
  		 "Encoding" : "thriftrw",
+         "TaskList"" : "taskList","
```
Comment (vancexu) [inline common/persistence/elasticsearch/decodeBench_test.go:46]: typo?

File: common/persistence/elasticsearch/decodeBench_test.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ var (
          "WorkflowID": "6bfbc1e5-6ce4-4e22-bbfb-e0faa9a7a604-1-2256",
          "WorkflowType": "TestWorkflowExecute",
  		 "Encoding" : "thriftrw",
```
Comment (vancexu) [inline common/persistence/elasticsearch/decodeBench_test.go:45]: nit: indent?

File: common/persistence/elasticsearch/esVisibilityStore.go [hunk 3]
```diff
@@ -910,13 +915,14 @@ func (v *esVisibilityStore) convertSearchResultToVisibilityRecord(hit *elastic.S
 
 func getVisibilityMessage(domainID string, wid, rid string, workflowTypeName string,
 	startTimeUnixNano, executionTimeUnixNano int64, taskID int64, memo []byte, encoding common.EncodingType,
-	searchAttributes map[string][]byte) *indexer.Message {
+	taskList string, searchAttributes map[string][]byte) *indexer.Message {
```
Comment (vancexu) [inline common/persistence/elasticsearch/esVisibilityStore.go:918]: nit: either put taskList to the end, or after workflowTypeName/taskID

**Comment by yycptt** [comment]
> > > I thought the original issue means include task list as one of the default search attributes (just like `BinaryCheckSums` and `CadenceChangeVersion`), not adding a new field to the visibility record. Am I missing something here?
> > > I am concerned about this because if we add a new field to the visibility record, it will impact visibility archival, especially our internal implementation.
> > 
> > 
> > The major reason is to also benefit users without ES.
> > We will need to continue adding this to visibility archival before release.
> 
> I see. Let's take the discussion offline and probably explore if there's an alternative. Making changes to visibility archival, especially our internal implementation will be a little bit tricky.

Synced offline. Will keep the current approach as tasklist is logically part of the workflow definition and should not be part of search attributes. Changes for visibility archival will be made in later PRs.

@mkolodezny Probably update the pr title to avoid any confusion. Probably `Add tasklist to visibility record`.

File: common/persistence/elasticsearch/decodeBench_test.go (hunk 2 above)
Comment (mkolodezny) [reply common/persistence/elasticsearch/decodeBench_test.go:45]: I do not see this in my editor for some reason.

File: config/dynamicconfig/development_es.yaml [hunk 4]
```diff
@@ -15,6 +15,7 @@ frontend.validSearchAttributes:
       CloseTime: 2
       CloseStatus: 2
       HistoryLength: 2
+      TaskList: 0
```
Comment (vancexu) [inline config/dynamicconfig/development_es.yaml:18]: 1 because its keyword type

**Pull request merged**